packages/markdown-docx/src/ToOOXMLVisitor/index.js

```diff
@@ -69,9 +77,10 @@ class ToOOXMLVisitor {
      * @param {string}  value           Text value of the node
      * @param {Array}   nodeProperties  Properties of the node
      * @param {boolean} calledByCode    Is function called by code node or not
+     * @param {object}  conditions      else and false conditions of optional and conditional nodes
      * @returns {string} Generated OOXML
      */
-    generateTextOrCodeOOXML(value, nodeProperties, calledByCode = false) {
+    generateTextOrCodeOOXML(value, nodeProperties, calledByCode = false, conditions) {
```

Comment:
Instead of introducing another `conditions attribute, we should combine its attributes, `isTraversingFalseCondition` and `isTraversingElseCondition`, with `nodeProperties` because their behaviour and function are essentially the same. You can probably create your own properties to distiguish between the above two attributes, whose difference is not clear to me.

Reply:
@algomaster99 
I think that conditions which I was thinking of renaming to `parentProps` will be better suited in this case. The reason being conditions applies to a set of nodes and subnodes. `Nodeinformation` is specific to a particular node. Also, it would make the nodeInformation complex.

In the list transformer we also need certain properties of parent, which will obviously increase the `nodeProperties` as we go down.